openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -23,202 +23,114 @@
-            creator = f"{first_name} {last_name}".strip()
-
-        if credit := result.get("credit"):
-            credit_line = credit.strip()
-            pub_date = result.get("submitted").strip()
-
-        return creator, credit_line, pub_date
+        TODO: Adapt `url` and `creator_url` to avoid redirects.
+        """
 
-    def get_record_data(self, data: dict) -> dict | list[dict] | None:
-        uid = data.get("uid")
+        uid = data.get("uuid")
         if not uid:
-            return
-        logger.debug(f"Processing UUID: {uid}")
-        params = {
-            "options": " ".join(
-                [
-                    "credit",
-                    "licenseURL",
-                    "pngFiles",
-                    "submitted",
-                    "submitter",
-                    "taxa",
-                    "canonicalName",
-                    "string",
-                    "firstName",
-                    "lastName",
-                ]
-            )
-        }
-        endpoint = f"{self.base_endpoint}/{uid}"
-        response_json = self.get_response_json(params, endpoint)
-        result = self._get_response_data(response_json)
-        if not result:
             return None
 
-        meta_data = {}
-        uid = result.get("uid")
-        license_url = result.get("licenseURL")
-
-        img_url, width, height = self._get_image_info(result, uid)
-
-        if img_url is None:
+        data = data.get("_links", {})
+        license_url = data.get("license", {}).get("href")
+        img_url = data.get("sourceFile", {}).get("href")
+        foreign_url = data.get("self", {}).get("href")
+        if not license_url or not img_url or not foreign_url:
             return None
 
-        meta_data["taxa"], title = self._get_taxa_details(result)
+        foreign_url = self.host + foreign_url
 
-        foreign_url = self._image_url(uid)
-
-        (
-            creator,
-            meta_data["credit_line"],
-            meta_data["pub_date"],
-        ) = self._get_creator_details(result)
+        title = data.get("self", {}).get("title")
+        creator = data.get("contributor", {}).get("title")
+        creator_url = self.host + data.get("contributor", {}).get("href")
+        width, height = self._get_image_sizes(data)
 
         return {
+            "license_info": get_license_info(license_url=license_url),
             "foreign_identifier": uid,
             "foreign_landing_url": foreign_url,
             "image_url": img_url,
-            "license_info": get_license_info(license_url=license_url),
+            "title": title,
+            "creator": creator,
+            "creator_url": creator_url,
             "width": width,
             "height": height,
-            "creator": creator,
-            "title": title,
-            "meta_data": meta_data,
+            # TODO: Evaluate whether to include upstream thumbnails.
+            # Sizes available: 192x192, 128x128, 64x64.
```

Comment:
I tried the other images we get in the `sample_record.json`: while the SVGs are all crisp on any resolution, the raster image of an acceptable size (https://images.phylopic.org/images/5b1e88b5-159d-495d-b8cb-04f9e28d2f02/raster/512x480.png) looks much worse. 

I wonder if we should handle SVGs differently: they can be basically any size, their width and height don't mean much beyond the aspect ratio (which of course is very important for the frontend rendering), and their thumbnails will probably be larger in size and worse in quality than the original image. Not for this PR specifically, of course.

Reply:
Yeah, the raster images are an option but look really blurry, not sure if it's worth including those for thumbnails.